File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.cpp
class Solution {
public:
    string addSpaces(string s, vector<int>& spaces) {
        string res = "";
        int j = 0;
        for(int i = 0;i<s.size();i++){
            if(j<spaces.size() && spaces[j] == i ){
                res+=" ";
                j+=1;
            }
            res+= s[i];
        }   

        return res;
    }
};

File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        res = ""
        j = 0
        for i in range(len(s)):
            if j<len(spaces) and i==spaces[j]:
                res+=" "
                j+=1
            res+=s[i]
        return res
                
            